node/core/pvf/src/host.rs

```diff
@@ -543,10 +595,29 @@ async fn handle_heads_up(
 				ArtifactState::Prepared { last_time_needed, .. } => {
 					*last_time_needed = now;
 				},
-				ArtifactState::Preparing { waiting_for_response: _ } => {
+				ArtifactState::Preparing { .. } => {
 					// The artifact is already being prepared, so we don't need to do anything.
 				},
-				ArtifactState::FailedToProcess(_) => {},
+				ArtifactState::FailedToProcess { last_time_failed, num_failures, error: _ } => {
+					// TODO: Do we want to retry for heads-up requests?
```

Comment:
Figured we wanted the same logic for heads-up requests, but wanted to double-check.